Automattic/newspack-blocks
# Read More Button

### All Submissions:






### Changes proposed in this Pull Request:

Implements "Load more" functionality for the Homepage Posts block.

There are 2 issues that still need to be addressed, but are not considered as blockers for V1 of this feature:

1. This feature is not yet AMP compatible and the button will be suppressed in AMP requests (see https://github.com/Automattic/newspack-blocks/pull/226#issuecomment-558695909)
2. The block has a post de-duplication feature that is not yet integrated in the Load More functionality. Under certain circumstances this may lead to showing duplicate posts when there are multiple instances of the block on a page. For more on this: `paYJgx-jW-p2`

### How to test the changes in this Pull Request:

- Add 2 or more `Homepage Posts` blocks via the editor (more than 1 block is needed to test [the state separation between them](https://github.com/Automattic/newspack-blocks/pull/226#pullrequestreview-324201713)).
- Go to block's `Display settings` and switch on the `Show "More" Button` toggle
- In the front-end view, you should see a `Load more articles` button under each block you've added
- When you click the button it should append more posts (the number of appended posts depends on the `Number of items` value set in related block's `Display settings`)
- When clicked, the button should be replaced with `Loading...` text
- When there are no more posts to load, the button should disappear
- When a request error occurs, a `Something went wrong. Please refresh the page and/or try again.` message should appear, but the button should stay visible to enable manual retries.

### Other information:




Closes: #154
